sdk/core/azure-core-amqp/inc/azure/core/amqp/message_sender.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "cancellable.hpp"
+#include "claims_based_security.hpp"
+#include "common/async_operation_queue.hpp"
+#include "connection.hpp"
+#include "link.hpp"
+#include "models/amqp_message.hpp"
+#include "models/amqp_value.hpp"
+#include <azure/core/nullable.hpp>
+#include <tuple>
+
+namespace Azure { namespace Core { namespace Amqp {
+  namespace _detail {
+    class MessageSenderImpl;
+  }
+  namespace _internal {
+    enum class MessageSendResult
+    {
+      Invalid,
+      Ok,
+      Error,
+      Timeout,
+      Cancelled
+    };
+    enum class MessageSenderState
+    {
+      Invalid,
+      Idle,
+      Opening,
+      Open,
+      Closing,
+      Error
+    };
+
+    enum class SenderSettleMode
+    {
+      Unsettled,
+      Settled,
+      Mixed
+    };
+
+    class MessageSender;
+    struct MessageSenderEvents
+    {
+      virtual void OnMessageSenderStateChanged(
+          MessageSender const& sender,
+          MessageSenderState newState,
+          MessageSenderState oldState)
+          = 0;
+    };
+
+    struct MessageSenderOptions
+    {
+      std::string Name;
+      SenderSettleMode SettleMode{};
+      std::string SourceAddress;
+      std::vector<std::string> AuthenticationScopes;
+      Azure::Nullable<uint64_t> MaxMessageSize;
+      bool EnableTrace{false};
+      Azure::Nullable<uint32_t> InitialDeliveryCount;
+
+      // Copied from Go, not sure if they're needed.
```

Comment:
Please add a link to the repo/source where these options are copied from.

Reply:
#4503